app/Jobs/Invoice/InvoiceWorkflowSettings.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/**
+ * Invoice Ninja (https://invoiceninja.com)
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2020. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://opensource.org/licenses/AAL
+ */
+
+namespace App\Jobs\Invoice;
+
+use App\Mail\Invoices\InvoiceWasPaid;
+use App\Models\Client;
+use App\Models\Invoice;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\Mail;
+
+class InvoiceWorkflowSettings implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    public $invoice;
+    public $client;
+
+    /**
+     * Create a new job instance.
+     *
+     * @return void
+     */
+    public function __construct(Invoice $invoice, Client $client = null)
+    {
+        $this->invoice = $invoice;
+        $this->client = $client ?? $invoice->client;
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        if ($this->client->getSetting('auto_archive_invoice')) {
+            $this->invoice->archive();
```

Comment:
I don't think this method exists on the model..

This method is implemented in the Base Repository